src/modules/70-pipeline/components/PipelineStudio/PipelineTemplateBuilder/TemplatePipelineSpecifications/TemplatePipelineSpecifications.tsx

```diff
@@ -0,0 +1,225 @@
+import { ErrorsStrip } from '@pipeline/components/ErrorsStrip/ErrorsStrip'
+import { useMutateAsGet } from '@common/hooks'
+import { yamlStringify } from '@common/utils/YamlHelperMethods'
+import css from './TemplatePipelineSpecifications.module.scss'
+
+export function TemplatePipelineSpecifications(): JSX.Element {
+  const {
+    state: { pipeline, schemaErrors, gitDetails },
+    allowableTypes,
+    updatePipeline,
+    isReadonly
+  } = usePipelineContext()
+  const queryParams = useParams<ProjectPathProps>()
+  const templateRef = getIdentifierFromValue(defaultTo(pipeline.template?.templateRef, ''))
+  const scope = getScopeFromValue(defaultTo(pipeline.template?.templateRef, ''))
+  const { showError } = useToaster()
+  const { getString } = useStrings()
+  const formikRef = React.useRef<FormikProps<unknown> | null>(null)
+  const formRefDom = React.useRef<HTMLElement | undefined>()
+  const [inputsTemplate, setInputsTemplate] = React.useState<PipelineInfoConfig>()
+  const [allValues, setAllValues] = React.useState<PipelineInfoConfig>()
+  const [initialValues, setInitialValues] = React.useState<PipelineInfoConfig>()
+  const [formikErrors, setFormikErrors] = React.useState<FormikErrors<PipelineInfoConfig>>()
+  const [showFormError, setShowFormError] = React.useState<boolean>()
+  const dummyPipeline = useRef(pipeline)
+
+  const onChange = React.useCallback(
+    debounce(async (values: PipelineInfoConfig): Promise<void> => {
+      await updatePipeline({ ...pipeline, ...values })
+    }, 300),
+    [pipeline, updatePipeline]
+  )
+
+  const {
+    data: pipelineResponse,
+    error: pipelineError,
+    refetch: refetchPipeline,
+    loading: pipelineLoading
+  } = useMutateAsGet(useGetYamlWithTemplateRefsResolved, {
+    queryParams: {
+      ...getScopeBasedProjectPathParams(queryParams, scope),
+      pipelineIdentifier: pipeline.identifier,
+      repoIdentifier: gitDetails.repoIdentifier,
+      branch: gitDetails.branch,
+      getDefaultFromOtherRepo: true
+    },
+    body: {
+      originalEntityYaml: yamlStringify(dummyPipeline.current)
+    }
+  })
+
+  const {
+    data: templateInputSetYaml,
+    error: templateInputSetError,
+    refetch: refetchTemplateInputSet,
+    loading: templateInputSetLoading
+  } = useGetTemplateInputSetYaml({
+    templateIdentifier: templateRef,
+    queryParams: {
+      ...getScopeBasedProjectPathParams(queryParams, scope),
+      versionLabel: defaultTo(pipeline.template?.versionLabel, ''),
+      repoIdentifier: gitDetails.repoIdentifier,
+      branch: gitDetails.branch,
+      getDefaultFromOtherRepo: true
+    }
+  })
+
+  React.useEffect(() => {
+    try {
+      const templateInputs = parse(defaultTo(templateInputSetYaml?.data, ''))
+      setInputsTemplate(templateInputs)
+      setFormikErrors({})
+    } catch (error) {
+      showError(error.message, undefined, 'template.parse.inputSet.error')
+    }
+  }, [templateInputSetYaml?.data])
```

Comment:
Codacy has a fix for the issue: [React Hook React.useEffect has a missing dependency: 'showError'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232314)
```suggestion
  }, [showError, templateInputSetYaml?.data])
```